lib/webidl2.js

```diff
@@ -8,13 +8,57 @@
     // against integers early.
     "float": /-?(?=[0-9]*\.|[0-9]+[eE])(([0-9]+\.[0-9]*|[0-9]*\.[0-9]+)([Ee][-+]?[0-9]+)?|[0-9]+[Ee][-+]?[0-9]+)/y,
     "integer": /-?(0([Xx][0-9A-Fa-f]+|[0-7]*)|[1-9][0-9]*)/y,
-    "identifier": /[A-Z_a-z][0-9A-Z_a-z-]*/y,
+    "identifier": /_?[A-Za-z][0-9A-Z_a-z-]*/y,
     "string": /"[^"]*"/y,
     "whitespace": /[\t\n\r ]+/y,
     "comment": /((\/(\/.*|\*([^*]|\*[^\/])*\*\/)[\t\n\r ]*)+)/y,
     "other": /[^\t\n\r 0-9A-Z_a-z]/y
   };
 
+  const namedTerminals = [
```

Comment:
nit: maybe we should alpha sort these. Might make it easier to maintain and spot duplicates, etc.